src/components/SideBar.vue

```diff
@@ -15,6 +20,31 @@ let pages = ['Home', 'Sports', 'Events', 'Teams', 'Preferences' ]
         :key="page.id"
         class="page"
       >
+        <IconHome
+          v-if="page === 'Home'"
+          class="icon"
+        />
+        
+        <IconSports
+          v-if="page === 'Sports'"
+          class="icon"
+        />
+        
+        <IconEvents
+          v-if="page === 'Events'"
+          class="icon"
+        />
+
+        <IconPrefs
+          v-if="page === 'Preferences'"
+          class="icon"
+        />
+
+        <IconTeams
+          v-if="page === 'Teams'"
+          class="icon"
+        />
```

Comment:
Remember to try using `<component>` to dynamically import rather than manually listing each of the components. Also, this will be part of your new sub-component as well.